### ISSUE TITLE
Empty Python lambda size is 3.4MB and in-line code editing on the Lambda console is impossible.

### ISSUE THREAD

**Issue by georgeleeventures** [aws-amplify/amplify-cli#9919]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### Amplify CLI Version

7.2.2 latest

### What operating system are you using?

Amazon Linux, Cloud9

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made.

### Amplify Categories

function

### Amplify Commands

add

### Describe the bug

Whenever I create a new Python lambda function and do an amplify push, the size of the lambda function is always over 3MB, usually 3.4MB. This disallows me to do code editing and testing in the lambda console which I am used to doing as we use Amplify with appsync and deploy lambda functions through Amplify. This occurs even when I create a blank default lambda function in python without any code changes or modifications to the default.

### Expected behavior

I should be able to make code edits and test on the lambda console because the size is usually under 3MB.

### Reproduction steps

1. amplify add function
2. Select python function
3. Make no changes and accept defaults
4. amplify push
5. Check lambda console and you get the message: The deployment package of your Lambda function "testfunc-dev" is too large to enable inline code editing. However, you can still invoke your function.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by josefaidt** [comment]
Hey @georgeleeventures :wave: thanks for raising this! As a follow-up question I wanted to clarify the motivation for editing directly in the Lambda Console rather than editing locally and testing with `amplify mock function`? Please keep in mind edits to the Lambda code in the Lambda Console will not be able to be "synced" locally with `amplify pull`

**Comment by georgeleeventures** [comment]
@josefaidt We use lambda layers. Last i checked, I cant mock the function with lambda layers... can I? I make edits in the lambda console and then once its working, I take it and re-deploy it in my amplify project. That is how I go about it right now because I need my lambda layers.

**Comment by josefaidt** [comment]
Hey @georgeleeventures thank you for clarifying, and you are correct `amplify mock` does not currently support Lambda Layers as documented here https://github.com/aws-amplify/amplify-cli/issues/9849. While we are unable to mitigate the size constraint for inline editing it sounds like you would also benefit from the feature request to enable function hot-swap, which is documented here https://github.com/aws-amplify/amplify-cli/issues/8642. Let me take a further look at this one and see if there's something we can do to cut down on the packaged size

**Comment by josefaidt** [comment]
Hey @georgeleeventures :wave: are you using Python 3.9 locally? I was able to reproduce the size issue using Python 3.9 locally, however if we use Python 3.8 (which the function's runtime is set to by default) the size should be around 2MB. Can you try downgrading to Python 3.8 and rebuild?

**Comment by georgeleeventures** [comment]
@josefaidt I ran python --version, and it seems like i have python 3.8.5. I don't think I'm using python 3.9. How can i check which runtime is being used by amplify?

**Comment by georgeleeventures** [comment]
I checked the pipfile info inside the lambda function and I am seeing Python 3.8 as the version. How come I am still having this issue? In it, I see:

[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]
src = {editable = true, path = "./src"}

[requires]
python_version = "3.8"

**Comment by georgeleeventures** [comment]
Bump on this

**Comment by georgeleeventures** [comment]
george:~/environment $ which python3
/usr/bin/python3
george:~/environment $ python3 --version
Python 3.8.5

**Comment by josefaidt** [comment]
Hey @georgeleeventures I'm taking a further look into this one to see if we can circumvent the large package size. My only concern would be once we add a dependency to the function it will again exceed the 3MB constraint from the Lambda Console. Is the motivation to edit the Lambda in place for rapid iteration and troubleshooting?

**Comment by georgeleeventures** [comment]
That's correct @josefaidt . I don't thinking adding a dependency should exceed the limit... I can use lambda layers for external libraries.

**Comment by Lacie** [comment]
I downloaded the deployment package from my lambda, and it seems that Amplify is [creating a deployment package using a venv](https://docs.aws.amazon.com/lambda/latest/dg/python-package.html#python-package-update-venv), effectively more than doubling the size of it.

@josefaidt Is there any particular reason Amplify is using a venv for dependencies and not just [adding them to the deployment package zip](https://docs.aws.amazon.com/lambda/latest/dg/python-package.html#python-package-create-package-with-dependency)? This is what I do for about half a dozen lambdas in our backend that require authentication via [requests-aws4auth](https://github.com/tedder/requests-aws4auth) to make GraphQL API calls. 

For comparison, I created a lambda in the console, and even with all its dependencies in the deployment package, the total size was less than 3MB. Adding that same lambda function via Amplify made it nearly 10MB. That's 7MB of bloat Amplify is adding, most of which is pip. I fail to see why it's necessary to include pip in a lambda deployment package

**Comment by josefaidt** [comment]
Hey @georgeleeventures and @Lacie apologies for the delay here! With @Lacie 's note regarding packaging `venv` and the comparison with deployed sizes of Python-based Lambda functions outside the context of Amplify I am going to mark this as a feature request to enhance the experience with Python functions 🙂

**Comment by prajendra-is** [comment]
Any update on this? I've been having the same issue. I reverted to python3.8, but just making a 'hello world' lambda results in a lambda that is too big for in-line code editing (3.9MB)

**Comment by fanando** [comment]
same issue here , is there an update on that matter?

**Comment by Vinod0511** [comment]
Any update on this? Is there any plan to include it as part of v11.x?

**Comment by emendoza2** [comment]
This is a much-needed feature. Would really like to know if there's been any movement on fixing it.

**Comment by Theantipioneer** [comment]
the most basic function from just running amplify add function already exceeds limit.

**Comment by mikecorlett** [comment]
Any update on this? 3+Mb for an empty function is crazy, apart from the inability to view the code in the console.
